lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,244 @@
+            '_', ' ').replace(':', ' ').split()
+
+        for name in fg_components:
+            fg_dict[name]
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def increment_is_valid(nn_taxon):
+    # check the increment is an int and has at least 2 digits
+    try:
+        assert int(nn_taxon)
+        assert len(nn_taxon) >= 2
+
+    except ValueError:
+        return False
+
+    else:
+        return True
+    
+
+def process_variable_is_valid(cspv_taxon):
+    # check control system process variable starts with a letter
+    match = re.search('^[a-zA-z]', cspv_taxon)
+    if match:
+        return True
+    else:
+        return False
+
+
+def validate(user_input):
+    try:
+        assert '.' not in user_input
+        assert (len(user_input) <= 60)
+
+    except AssertionError:
+        print('Invalid')
+        return False
+
+    pv_name = [x.strip() for x in user_input.split(':')]
+
+    for element in pv_name:
+        if not re.match('^[a-zA-Z0-9]+', element):
+            print('Invalid')
+            return False
+
+    # Check the length of the PV name is valid
+    if (len(pv_name) < 3) or (len(pv_name) > 5):
+        print('Invalid')
+        return False
+
+    # parse the functional component (FFFFF)
+    fc_taxon = list(pv_name[0].strip())
+    fc_valid = functional_component_is_valid(fc_taxon)
+
+    if fc_valid:
+
+        # check for PV name with 3 elements (FFFFF:CCC:XXXX)
+        if (len(pv_name) == 3):
+            ccc_taxon = pv_name[1].strip()
+            ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+            if ccc_valid:
+                print('Valid')
+                return True
+            else:
+                print('Invalid')
+                return False
+
+        # Check for PV name with 4 elements (FFFFF:GGG:CCC:XXXX or FFFFF:CCC:NN:XXXX).
+        if (len(pv_name) == 4):
+            fg_taxon = pv_name[1].strip()
+            fg_valid = fungible_is_valid(fg_taxon)
+
+            if fg_valid:
+                # validate for FFFFF:GGG:CCC:XXXX
+                ccc_taxon = pv_name[2].strip()
+                ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+                cspv_taxon = pv_name[3].strip()
+                cspv_valid = process_variable_is_valid(cspv_taxon)
+                              
+                if ccc_valid and cspv_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+            else:
+                # validate for FFFFF:CCC:NN:XXXX
+                ccc_taxon = pv_name[1].strip()
+                ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+                nn_taxon = pv_name[2].strip()
+                nn_valid = increment_is_valid(nn_taxon)
+
+                if ccc_valid and nn_valid:
+                    print('Valid')
+                    return True
```

Comment:
I think XXXX checks may have been skipped in this case too

Reply:
Same for this one